### PR TITLE
fix inconsistent expandFilename on Windows

### PR DESCRIPTION
**Pull request by nc-x**
Closes https://github.com/nim-lang/Nim/issues/10153

### CODE REVIEW

File: compiler/msgs.nim [hunk 1]
```diff
@@ -67,13 +67,14 @@ proc fileInfoKnown*(conf: ConfigRef; filename: AbsoluteFile): bool =
     canon = filename
   result = conf.m.filenameToIndexTbl.hasKey(canon.string)
 
-proc fileInfoIdx*(conf: ConfigRef; filename: AbsoluteFile; isKnownFile: var bool): FileIndex =
+proc fileInfoIdx*(conf: ConfigRef; filename: AbsoluteFile; isKnownFile: var bool, canonicalize: bool = true): FileIndex =
   var
-    canon: AbsoluteFile
+    canon = filename
```
Comment (Araq) [inline compiler/msgs.nim]: Would be nice if we could avoid this string copy somehow.

**Comment by nc-x** [comment]
So the code already caters for this use case, but somehow the exception is not being raised (IDK if it worked previously)

https://github.com/nim-lang/Nim/blob/a1e268e3dccdde4df9b11a0ee87971e1143fbb43/compiler/msgs.nim#L75-L82

**Comment by nc-x** [comment]
Fixed the issue differently - canonicalizePath now checks if file exists (on windows). It was already the case for other OSes because they use `realpath` which AFAIK already does the check.

**Comment by Araq** [comment]
Very nice, but I think we should fix ``os.nim`` instead so that it behaves consistently across platforms.

**Comment by nc-x** [comment]
https://github.com/nim-lang/Nim/blob/a1e268e3dccdde4df9b11a0ee87971e1143fbb43/compiler/options.nim#L484-L494

According to this comment, expandFilename already has different behaviour on windows

So, should I move these two lines over to the expandFilename implementation as well?
https://github.com/nim-lang/Nim/blob/a1e268e3dccdde4df9b11a0ee87971e1143fbb43/compiler/options.nim#L491-L492

**Comment by Araq** [comment]
> So, should I move these two lines over to the expandFilename implementation as well?

Hmm, yes, I think so.

**Comment by timotheecour** [comment]
/cc @araq @nc-x  for anyone interested : the github diff is huge but `git show --color-moved` shows exactly what was just moved(and where), vs what was added/removed
```
git show --color-moved e9a192c36fb62331db0bbd308433e56fc771c352
```

**Pull request merged**